dev_docs/key_concepts/security.mdx

```diff
@@ -0,0 +1,101 @@
+1. Performs non-insignificant processing, causing load on the Kibana server.
+2. Calls Elasticsearch using the internal `kibana_system` user.
+3. Calls a third-party service.
+4. Returns any non-public information to the caller, such as system configuration or state.
+
+More information on adding `access` tags to your routes can be found temporarily in the [legacy documentation](https://www.elastic.co/guide/en/kibana/current/development-security.html#development-plugin-feature-registration)
+
+### Why not add `access` tags to all routes by default?
+Each authorization check that we perform involves a round-trip to Elasticsearch, so they are not as cheap as we'd like. Therefore, we want to keep the number of authorization checks we perform within a single route to a minimum.
+Adding an `access` tag to routes that leverage the Saved Objects Service would be redundant in most cases, since the Saved Objects Service will be performing authorization anyway.
+
+
+## The `kibana_system` user
+
+The Kibana server authenticates to Elasticsearch using the `elastic/kibana` [service account](https://www.elastic.co/guide/en/elasticsearch/reference/current/service-accounts.html#service-accounts-explanation). This service account has privileges that are equivilent to the `kibana_system` reserved role, whose descriptor is managed in the Elasticsearch repository ([source link](https://github.com/elastic/elasticsearch/blob/56633e0a2934b03090ab726c33e5289129c98bfe/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java#L682)).
+The vast majority of features will not require changes to the `kibana_system` user privileges. Changes to these privileges must be carefully considered, as we do not want the `kibana_system` account to have access to user data.
+
+### Guiding principals
+
+Consider these guidelines when reviewing changes to the `kibana_system` role descriptor.
+
+#### 1. Kibana should only access system and hidden indices
+
+- System indices are managed entirely by the stack, and should never be accessed by end users.
+- Hidden indices are _typically_ managed by the stack, but _may_ be accessed by end users.
+- Data indices are those which an end user could conceivably create on their own. As a general rule, users can create indices of any pattern so long as it does not begin with a \``.`\` (dot). Users can also create hidden indices, so it is important that documentation exists for any stack-managed hidden indices to reduce the chance of conflict with user-managed indices.
+
+Therefore, Kibana should not have access to non-system indices which do not begin with a \``.`\` (dot).
+
+Kibana should also not have the ability to modify system/hidden indices for which it is not the owner.
+
+##### Examples
+|Pattern|Allowed?|Notes|
+|-------|--------|-----
+|`my-data`|No| User-defined data index.
+|`kibana-metrics`|No| User-defined data index.
+|`.security`|No| System index not owned by Kibana.
```

Comment:
I'm not sure this is accurate. We have loads of system indices like this in the `kibana_system` role descriptor that grant `all` or `write` access. Why are they allowed?

Reply:
`.reporting-*`, `.management-beats`, `.apm-agent-configuration`, `.apm-custom-link`, and `.apm-source-map` are all managed by Kibana as far as I can tell.

`.fleet` _may_ be an exception. Fleet was originally 100% within Kibana, so it may have started out as a Kibana-owned index. Parts of Fleet were then extracted into a dedicated Fleet Server, so this index likely has shared ownership.

Reply:
Ah got you, sorry, I got confused thinking that the table was about the naming convention since it implies that an index pattern like `.security` wouldn't be allowed since it's not owned by Kibana whereas a pattern starting with `.kibana*` would be allowed since that denotes that it's owned by Kibana. 

I would swap the table around to make clear that it's about the type of index, not the index pattern:

| Index Type | Allowed Permissions | Examples |
|-------|--------|-----
| User-defined data index | none | `my-data`, `kibana-metrics` |
| System index not owned by Kibana | `read` | `.security` |
| System indices owned by Kibana | `all` | `.kibana*` |

Reply:
Thanks, that's a great suggestion. Will make this change 👍

Reply:
Addressed in https://github.com/elastic/kibana/pull/161697/commits/24d7aa5c1fca8e1798e3b3f4a7706eb25ea948bf